diffpy/snmf/subroutines.py

```diff
@@ -84,6 +84,35 @@ def construct_stretching_matrix(components, number_of_components, number_of_sign
     return stretching_factor_matrix
 
 
+def construct_component_matrix(components, number_of_components, signal_length):
```

Comment:
we don't need to pass in `number_of_components`  because that is len(components).  I fixed this in the other function before merging, but looking back I see that I did all the t hings except remove it from the signature.   Please can you remove it from the arguments here and and also there?  Unless I have missed something....

Reply:
Ok. I will fix that here and in the other function